### DISCUSSION TITLE
Longstanding need for improvements and refactoring

### DISCUSSION THREAD

**Discussion by GeneMosher** [ViewTouch/viewtouch#136] [Ideas]
I would like to see AI invoked in a refactoring of the font display code and the touchscreen input code.  
The font display code really should not be relying on bit-mapped fonts; it should instead be relying on scalable outline fonts.
The touchscreen input code should be redone to make use of the latest xinput code and no longer interpret mouse clicks as touches.
Anyone interested in working on any of these three issues should reply here or just call me (Gene) at 541-515-5913 (USA).

**Comment by muskrat7** [comment]
I read an old post discussing GTK. Has it been migrated to this yet? Curious on the progress of this project as I am looking at starting a bar soon and would like to use viewtouch as the POS.

**Reply by GeneMosher** [reply]
The use of GTK was halted and the ViewTouch dependencies on GTK were removed.  ViewTouch graphics are achieved with X Window System Primatives and have never made use of GTK.  If you're a C++ programmer I can give you a couple of suggestions which would improve ViewTouch PoS and shouldn't be difficult to implement, depending on your level of experience.  Running a bar can be fun and profitable !  I'm a former restaurateur myself, naturally.  There are many hospitality locations using ViewTouch, some of them for over 20 years.  It's in quite good shape, in any case, however, it does not contain any EFT capabilities.  You simply make use of any EFT solution of your choice.  I highly recommend that you use the ViewTouch all-in-one shown on the ViewTouch Website home page and Bixolon printers/cash drawers.  Wired Ethernet is best but Wireless works just fine, too.  I can offer you Bar interfaces which are in use in many other ViewTouch locations.  Where are you located?

**Reply by muskrat7** [reply]
I do know C/C++ along with other languages but it has been a while since I have done anything. I would love to help though. I'm in Wisconsin and will send you an email regarding the solution. I definitely want to use ViewTouch and would like recommendations on the other items I need to complete my setup.

**Comment by GeneMosher** [comment]
The control of the X Windows System by the same people at Red Hat who are behind the development of Wayland has been exposed as a scheme not in the best interest of the X Windows System code and its users.  My take on this is that the Xlibre fork is a good and necessary development in the long history of X.  I fully expect that the Xlibre fork must replace Xorg (itself also a fork) as soon as possible.

**Reply by NeroBurner** [reply]
I think moving on from X would be better than trying to cling to a dead/dying technology. Wayland is done by the XOrg maintainers. No-one wanted to maintain XOrg anymore. The way forward is Wayland.

**Comment by GeneMosher** [comment]
The first AI enhancements to the ViewTouch source code are taking place now.  The significance of this cannot be overstated.

**Comment by GeneMosher** [comment]
We need to ensure remote display functionality is maintained.  Red HAT
developers run Wayland politics.  After 10 years of Wayland do we have
solid and proven remote display functionality?
